docs/contributors/opentracingprovider.adoc

```diff
@@ -0,0 +1,203 @@
+
+
+
+QUESTION: [Batch size/interval - should each reporter support this, or should it be a common capability independent of the transport?]
+
+
+
+== Mapping Span to APM Node
+
+=== Processing References
+
+When requested to create a Span, the first step is to assess the supplied list of References (including ‘ChildOf’ references that may be provided separate from other e.g. ‘FollowsFrom’ references).
+
+==== Identify a Primary Reference
+
+A primary reference is the one that is considered the true ‘parent’ to the Span being created. It may be extracted SpanContext, a parent Span or a referenced to a FollowsFrom Span. Each will be handled in a different way - but first we will discuss how to identify whether a primary reference exists.
+
+
+First step is to identify the number of each type of reference, i.e.
+
+(a) References representing Extracted Trace State, whether ref type is ChildOf or FollowFrom
+
+(b) ChildOf references representing internal Parent/Child Span relationship -> childOf(apmSpan)
+
+(c) FollowsFrom references representing internal links to previous related Spans
+
+
+Once we have these numbers, we use the following logic to determine whether a primary reference can be determined:
+
+* If references of type (a) exist, then if there is one reference it will be the primary. If there is more than one, then there is no distinct primary.
+
+* If there are no (a) references, then we check references of type (b) to see if they exist, and if so, if there is only one then it will be the primary. As previously, if there is more than one, then there is no distinct primary.
+
+* Finally if there are no (a) or (b) references, then we would check references of type (c). If one exists, it will be the primary, otherwise the is no primary.
+
+
+This approach means that references related to inbound communication (i.e. invocation of a service) take priority in terms of building a trace - which should be natural considering our principal aim is to build a distributed trace of communications between services.
+
+==== Processing a Primary Reference
+
+If a primary reference is found, then its processing will be dependent upon its type (i.e. a, b or c from previous section).
+
+===== Reference representing Extracted Trace State
+
+This reference represents potential extracted trace state propagated from an invoking client. Even if no trace state is actually found, it indicates that a communication has occurred to trigger this service.
+
+Therefore a new Trace fragment (and context) should be started, with the trace state (if available) initialised in the context and the top level Span should be represented by a Consumer node within that Trace fragment. The ‘id’ part of the trace state should be registered as a correlation id (of scope Interaction) on the Consumer.
+
+The other references should then be processed as discussed in the “Processing Remaining References” section below.
```

Comment:
Might be good to say that those added correlation ids will be added on newly added node. In case of FollowsFrom is it a consumer or following Component?